### ISSUE TITLE
YAML merge key anchors cause `json-schema-validator/no-invalid`

### ISSUE THREAD

**Issue by Kenneth-Sills** [ota-meshi/eslint-plugin-json-schema-validator#500]
A [YAML merge key](https://yaml.org/type/merge.html) allows expanding anchored objects into a new object definition. Since this plugin validates the schema, I would expect that it would recognize the anchor, merge the keys, and then evaluate schema correctness.

However, right now, the object being merged into receives the error `json-schema-validator/no-invalid` instead. This requires ignoring the rule here to avoid the error, as well as prevents proper schema validation in the upstream object being merged from. If the upstream object includes required keys, this rule may need to be entirely disable for the file.

It's a matter of taste if a corrected implementation should report on the line of the merge key itself or on the property definition in the object being merged, or how that information could be presented between e.g.:

```yaml
# example.yaml
# Expected schema:  [ { y: int } ]
- &CENTER { x: 1, y: 2 }
- << : *CENTER
```

> Line 2: Attempted to merge object with unexpected property `x`.

And:

> Line 1: Unexpected property `x` in object being merged.
